saleor/graphql/checkout/mutations/checkout_lines_update.py

```diff
@@ -1,15 +1,44 @@
 import graphene
+from django.forms import ValidationError
 
+from ....checkout.error_codes import CheckoutErrorCode
 from ....warehouse.reservations import is_reservation_enabled
-from ...core.types import CheckoutError
+from ...core.descriptions import DEPRECATED_IN_3X_INPUT
+from ...core.scalars import UUID
+from ...core.types import CheckoutError, NonNullList
 from ..types import Checkout
+from .checkout_create import CheckoutLineInput
 from .checkout_lines_add import CheckoutLinesAdd
 from .utils import check_lines_quantity
 
 
+class CheckoutLineUpdateInput(CheckoutLineInput):
+    quantity = graphene.Int(
+        required=False,
+        description="The number of items purchased. Required for not app requestor.",
```

Comment:
This might sound more natural:

The number of items purchased. Optional for apps, required for any other users.